### ISSUE TITLE
task txg_sync: blocked for more than 120 seconds

### ISSUE THREAD

**Issue by snsb-seifert** [zfsonlinux/zfs#4361]
Hello,

I tried to update to **ZFS 0.6.5.4** (from  0.6.4.2) on Gentoo 4.4.1 (from 4.0.5) but when working with the pools the commands get stuck in process state D and do not recover (even after 24h). There is no disk or cpu activity.

I tested this on different machines and also on Gentoo 4.1.15-r1. 
I get these errors at different commands. Sometimes when I import the pool (zpool import) or zfs receive. Currently I can reproduce it when I want to destroy the filesystems (zfs destroy -R ).

The systems are not use able as ZFS blocks.

Any Ideas?

[modinfo-zfs.txt](https://github.com/zfsonlinux/zfs/files/141997/modinfo-zfs.txt)
[modinfo-spl.txt](https://github.com/zfsonlinux/zfs/files/141998/modinfo-spl.txt)
[arcsummary.txt](https://github.com/zfsonlinux/zfs/files/141999/arcsummary.txt)

The log entries for **zfs destroy** are are:

```
kernel: INFO: task txg_sync:31838 blocked for more than 120 seconds.
kernel:       Tainted: P           O    4.4.1-gentoo #3
kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
kernel: txg_sync        D ffff8800a761fb88 13352 31838      2 0x00000000
kernel:  ffff8800a761fb88 ffff8807556a9840 ffffffff82214540 ffff8807533e9a80
kernel:  ffff8800a7620000 ffff8807556a97d8 ffff8807556a9840 ffff8807556a9818
kernel:  0000000000000000 ffff8800a761fba0 ffffffff81b1b4d7 ffff8807556a9838
kernel: Call Trace:
kernel:  [<ffffffff81b1b4d7>] schedule+0x37/0x80
kernel:  [<ffffffffa01c7754>] __cv_timedwait_sig+0xf4/0x120 [spl]
kernel:  [<ffffffff810e5840>] ? wait_woken+0x90/0x90
kernel:  [<ffffffffa01c7790>] __cv_wait+0x10/0x20 [spl]
kernel:  [<ffffffffa0678ff7>] rrw_enter_write+0x37/0xa0 [zfs]
kernel:  [<ffffffffa067906e>] rrw_enter+0xe/0x20 [zfs]
kernel:  [<ffffffffa0670268>] dsl_sync_task_sync+0x88/0x290 [zfs]
kernel:  [<ffffffffa06685a3>] dsl_pool_sync+0x313/0x410 [zfs]
kernel:  [<ffffffffa06820f2>] spa_sync+0x342/0x10e0 [zfs]
kernel:  [<ffffffff810e584d>] ? autoremove_wake_function+0xd/0x30
kernel:  [<ffffffffa069325e>] txg_fini+0x5de/0xc20 [zfs]
kernel:  [<ffffffffa0692ea0>] ? txg_fini+0x220/0xc20 [zfs]
kernel:  [<ffffffffa01c2cd7>] __thread_exit+0x97/0xb0 [spl]
kernel:  [<ffffffffa01c2c60>] ? __thread_exit+0x20/0xb0 [spl]
kernel:  [<ffffffff810c92f1>] kthread+0xe1/0x100
kernel:  [<ffffffff810c9210>] ? kthread_create_on_node+0x190/0x190
kernel:  [<ffffffff81b2045f>] ret_from_fork+0x3f/0x70
kernel:  [<ffffffff810c9210>] ? kthread_create_on_node+0x190/0x190
kernel: INFO: task zfs:981 blocked for more than 120 seconds.
kernel:       Tainted: P           O    4.4.1-gentoo #3
kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
kernel: zfs             D ffff88075b557ba0 13936   981  29993 0x00000000
kernel:  ffff88075b557ba0 ffff8807556a9398 ffff88080c26a7c0 ffff8807f8d24f80
kernel:  ffff88075b558000 ffff8807556a9290 ffff8807556a9398 ffff8807556a92d0
kernel:  0000000000000000 ffff88075b557bb8 ffffffff81b1b4d7 ffff8807556a9390
kernel: Call Trace:
kernel:  [<ffffffff81b1b4d7>] schedule+0x37/0x80
kernel:  [<ffffffffa01c7754>] __cv_timedwait_sig+0xf4/0x120 [spl]
kernel:  [<ffffffff810e5840>] ? wait_woken+0x90/0x90
kernel:  [<ffffffffa01c7790>] __cv_wait+0x10/0x20 [spl]
kernel:  [<ffffffffa06928d4>] txg_wait_synced+0xe4/0x130 [zfs]
kernel:  [<ffffffffa06e96c0>] ? dsl_destroy_snapshot_check_impl+0x90/0x190 [zfs]
kernel:  [<ffffffffa067000f>] dsl_sync_task+0x16f/0x260 [zfs]
kernel:  [<ffffffffa06e8970>] ? dsl_destroy_snapshot_sync_impl+0xb40/0xc30 [zfs]
kernel:  [<ffffffffa06e96c0>] ? dsl_destroy_snapshot_check_impl+0x90/0x190 [zfs]
kernel:  [<ffffffffa06e8970>] ? dsl_destroy_snapshot_sync_impl+0xb40/0xc30 [zfs]
kernel:  [<ffffffffa06e983b>] dsl_destroy_snapshots_nvl+0x7b/0xd0 [zfs]
kernel:  [<ffffffffa06bd415>] zfs_unmount_snap+0x245/0x270 [zfs]
kernel:  [<ffffffffa06bd439>] zfs_unmount_snap+0x269/0x270 [zfs]
kernel:  [<ffffffffa06bb0e7>] zfs_secpolicy_smb_acl+0x6377/0x6440 [zfs]
kernel:  [<ffffffffa06bad82>] zfs_secpolicy_smb_acl+0x6012/0x6440 [zfs]
kernel:  [<ffffffff811a92bc>] ? vma_merge+0x1fc/0x330
kernel:  [<ffffffff811eab3c>] do_vfs_ioctl+0x2fc/0x550
kernel:  [<ffffffff811eae04>] SyS_ioctl+0x74/0x80
kernel:  [<ffffffff81b20117>] entry_SYSCALL_64_fastpath+0x12/0x6a
```

**Comment by kdomanski** [comment]
Same issue here on CoreOS built with ZoL.

**Comment by behlendorf** [comment]
@snsb-seifert if you're able to reproduce this problem could you dump all the stack traces from a hung system and post them.  The `zfs destroy` command is correctly waiting for the `txg_sync` thread to finish destroying the snapshots.  And the `txg_sync` thread is blocked waiting on the `dp_config_rwlock` lock but it's not at all clear what process is holding it.  Hopefully the rest of the stack traces will shed some light on this.

```
echo t >/proc/sysrq-trigger
dmesg
```

**Comment by snsb-seifert** [comment]
Here are the stack traces.
[log.txt](https://github.com/zfsonlinux/zfs/files/152253/log.txt)

I did the following:
**zfs import thepool**
**zfs destroy -R thepool/thefilesystem**
zfs destroy is now in D-state
zpool status is callable and shows that everything is ok.

**Comment by kdomanski** [comment]
@behlendorf For me the issue is no longer present when running master.

**Comment by snsb-seifert** [comment]
Upgraded to **MASTER**
ZFS 513168abd267f84ebcb688b4e1d42a2abd4fedce
SPL 18d2f56176494a14437f18cd8482d151d634cc18
it is **working again**.
_gentoo:_ sys-fs/zfs-9999, sys-kernel/spl-9999, ys-fs/zfs-kmod-9999

zfs destroy immediately starts working on the disk and exits normally.
zfs receive is working normally.

For testing I returned back to 0.6.5.4 (emerge). Again txg_sync and zfs are in state D waiting, without cpu-load or disk access. zfs destroy does nothing to the file-systems.
Also zfs receive does not finish but waits at the end of the stream for something in state D, but file-systems seem to perfect.
This is like on the other machines where even zfs import is stuck in state D after the import of a pool. 
_gentoo:_ sys-kernel/spl-0.6.5.4-r1, sys-fs/zfs-kmod-0.6.5.4-r1, sys-fs/zfs-0.6.5.4-r2

What could be changed since the 0.6.5.4 release?

**Comment by orivej** [comment]
The release is probably fine, it is gentoo ebuilds that bring patches not present in the release. You may try versions without `-r` suffix, they are not patched. I had the same or similar issue #4347.

**Comment by kdomanski** [comment]
@behlendorf Experiencing the same issue again with both 0.6.5.5 and 0.6.5.6 without any additional patches.

```
[ 1801.170921] INFO: task sync:14705 blocked for more than 120 seconds.
[ 1801.178501]       Tainted: P           OE   4.4.6-coreos #2
[ 1801.185236] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[ 1801.194335] sync            D ffff88015e25be08     0 14705  14696 0x00000084
[ 1801.202668]  ffff88015e25be08 000000015e25bed8 ffff88040b985640 ffff88033a445640
[ 1801.211353]  ffff88015e25c000 ffff880408e82070 ffff880408e82088 ffff880408e82070
[ 1801.219975]  ffffffffac2050e0 ffff88015e25be20 ffffffffac568015 ffff88033a445640
[ 1801.228591] Call Trace:
[ 1801.232187]  [<ffffffffac2050e0>] ? SyS_tee+0x3c0/0x3c0
[ 1801.238598]  [<ffffffffac568015>] schedule+0x35/0x80
[ 1801.244724]  [<ffffffffac56ae71>] rwsem_down_read_failed+0x101/0x160
[ 1801.252232]  [<ffffffffac2e3b44>] call_rwsem_down_read_failed+0x14/0x30
[ 1801.260024]  [<ffffffffac56a410>] ? down_read+0x20/0x30
[ 1801.266466]  [<ffffffffac1d52c2>] iterate_supers+0xa2/0x110
[ 1801.273151]  [<ffffffffac205454>] sys_sync+0x44/0xb0
[ 1801.279256]  [<ffffffffac56c3ae>] entry_SYSCALL_64_fastpath+0x12/0x71
[ 1801.286883] INFO: task sync:17940 blocked for more than 120 seconds.
[ 1801.294378]       Tainted: P           OE   4.4.6-coreos #2
[ 1801.301164] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[ 1801.310186] sync            D ffff8803f3097e08     0 17940  14696 0x00000084
[ 1801.318461]  ffff8803f3097e08 00000001f3097ed8 ffff88040b983980 ffff8803a8383980
[ 1801.327076]  ffff8803f3098000 ffff880408e82070 ffff880408e82088 ffff880408e82070
[ 1801.335970]  ffffffffac2050e0 ffff8803f3097e20 ffffffffac568015 ffff8803a8383980
[ 1801.344660] Call Trace:
[ 1801.348263]  [<ffffffffac2050e0>] ? SyS_tee+0x3c0/0x3c0
[ 1801.354732]  [<ffffffffac568015>] schedule+0x35/0x80
[ 1801.360828]  [<ffffffffac56ae71>] rwsem_down_read_failed+0x101/0x160
[ 1801.368346]  [<ffffffffac2e3b44>] call_rwsem_down_read_failed+0x14/0x30
[ 1801.376126]  [<ffffffffac56a410>] ? down_read+0x20/0x30
[ 1801.382523]  [<ffffffffac1d52c2>] iterate_supers+0xa2/0x110
[ 1801.389248]  [<ffffffffac205454>] sys_sync+0x44/0xb0
[ 1801.395353]  [<ffffffffac56c3ae>] entry_SYSCALL_64_fastpath+0x12/0x71
[ 1801.402953] INFO: task zfs:18061 blocked for more than 120 seconds.
[ 1801.410403]       Tainted: P           OE   4.4.6-coreos #2
[ 1801.417139] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[ 1801.426153] zfs             D ffff88033e5fb8f8     0 18061   5762 0x00000080
[ 1801.434408]  ffff88033e5fb8f8 0000000000000286 ffff8803c369b980 ffff8803f4395640
[ 1801.443075]  ffff88033e5fc000 0000000000000121 ffffffffc07861e0 ffffffffc0788ba8
[ 1801.451733]  ffff88040a835000 ffff88033e5fb910 ffffffffac568015 ffffffffc0787b40
[ 1801.460418] Call Trace:
[ 1801.463985]  [<ffffffffac568015>] schedule+0x35/0x80
[ 1801.470151]  [<ffffffffc076e400>] spl_panic+0xf0/0x100 [spl]
[ 1801.476996]  [<ffffffffc076881d>] ? spl_kmem_free+0x3d/0x50 [spl]
[ 1801.484334]  [<ffffffffc0784687>] ? fnvlist_remove_nvpair+0x97/0x2aa [znvpair]
[ 1801.492810]  [<ffffffffc077e77e>] ? nvlist_free+0x8e/0xb0 [znvpair]
[ 1801.500320]  [<ffffffffc076881d>] ? spl_kmem_free+0x3d/0x50 [spl]
[ 1801.507619]  [<ffffffffc0784687>] ? fnvlist_remove_nvpair+0x97/0x2aa [znvpair]
[ 1801.516121]  [<ffffffffc07805d4>] ? nvlist_remove+0x3b4/0x6c0 [znvpair]
[ 1801.523984]  [<ffffffffc0784570>] fnvlist_add_nvlist_array+0x70/0x80 [znvpair]
[ 1801.532401]  [<ffffffffc0b3fbe9>] vdev_config_generate+0x549/0x620 [zfs]
[ 1801.540308]  [<ffffffffc0768deb>] ? spl_kmem_alloc_impl+0xbb/0x190 [spl]
[ 1801.548220]  [<ffffffffc0768deb>] ? spl_kmem_alloc_impl+0xbb/0x190 [spl]
[ 1801.556128]  [<ffffffffc0b2b3b1>] spa_config_generate+0x261/0x910 [zfs]
[ 1801.563908]  [<ffffffffc0b26893>] spa_lookup_by_guid+0x23c3/0x2790 [zfs]
[ 1801.571795]  [<ffffffffc0b26d23>] spa_get_stats+0x63/0x560 [zfs]
[ 1801.578964]  [<ffffffffac1654cf>] ? alloc_kmem_pages_node+0x4f/0xd0
[ 1801.586432]  [<ffffffffac1b266e>] ? kmalloc_large_node+0x2e/0x60
[ 1801.593601]  [<ffffffffac1b7227>] ? __kmalloc_node+0x257/0x300
[ 1801.600623]  [<ffffffffc0768c52>] ? spl_kmem_zalloc+0xb2/0x190 [spl]
[ 1801.608164]  [<ffffffffc0b5c708>] zfs_secpolicy_smb_acl+0x19a8/0x6110 [zfs]
[ 1801.616315]  [<ffffffffc0b5f407>] zfs_secpolicy_smb_acl+0x46a7/0x6110 [zfs]
[ 1801.624508]  [<ffffffffac2754af>] ? selinux_file_ioctl+0x10f/0x1c0
[ 1801.631841]  [<ffffffffac1e5294>] do_vfs_ioctl+0x2c4/0x4a0
[ 1801.638508]  [<ffffffffac110201>] ? __audit_syscall_entry+0xb1/0x100
[ 1801.646028]  [<ffffffffac26d213>] ? security_file_ioctl+0x43/0x60
[ 1801.653247]  [<ffffffffac1e54e9>] SyS_ioctl+0x79/0x90
[ 1801.659437]  [<ffffffffac56c3ae>] entry_SYSCALL_64_fastpath+0x12/0x71
```

**Comment by behlendorf** [comment]
@kdomanski the last issue you reported is different and is the result of hitting an assertion in the code.  Notice the `spl_panic` line in the stack trace from your last comment.  Are you able to consistently reproduce this?  If so what command are you running?  There should be an additional line in the console log showing the panic.

**Comment by kdomanski** [comment]
@behlendorf I'm able to consistently reproduce the two first stack traces, but not the third one. Basically, whenever I have some stale mountpoints or exclusively locked files under a ZFS mountpoint, then ZFS either fails to unmount or (more often) unmounts fine but gives me these stack traces on a reboot/shutdown, effectively locking the machine right before reboot.

I get "Reached target Shutdown" and then the machine spits out the shorter stack trace every 2 minutes.
Tried both on latest release and master.

**Comment by kdomanski** [comment]
@behlendorf One interesting phenomenon I noticed is a huge amount of kernel threads spawned by ZFS. Here's a full list. Is that normal?

```
  PID TTY          TIME CMD
    2 ?        00:00:00 kthreadd
    3 ?        00:00:00 ksoftirqd/0
    5 ?        00:00:00 kworker/0:0H
    6 ?        00:00:00 kworker/u32:0
    7 ?        00:00:00 rcu_sched
    8 ?        00:00:00 rcu_bh
    9 ?        00:00:00 migration/0
   10 ?        00:00:00 watchdog/0
   11 ?        00:00:00 watchdog/1
   12 ?        00:00:00 migration/1
   13 ?        00:00:00 ksoftirqd/1
   15 ?        00:00:00 kworker/1:0H
   16 ?        00:00:00 watchdog/2
   17 ?        00:00:00 migration/2
   18 ?        00:00:00 ksoftirqd/2
   20 ?        00:00:00 kworker/2:0H
   21 ?        00:00:00 watchdog/3
   22 ?        00:00:00 migration/3
   23 ?        00:00:00 ksoftirqd/3
   25 ?        00:00:00 kworker/3:0H
   26 ?        00:00:00 watchdog/4
   27 ?        00:00:00 migration/4
   28 ?        00:00:00 ksoftirqd/4
   30 ?        00:00:00 kworker/4:0H
   31 ?        00:00:00 watchdog/5
   32 ?        00:00:00 migration/5
   33 ?        00:00:00 ksoftirqd/5
   35 ?        00:00:00 kworker/5:0H
   36 ?        00:00:00 watchdog/6
   37 ?        00:00:00 migration/6
   38 ?        00:00:00 ksoftirqd/6
   40 ?        00:00:00 kworker/6:0H
   41 ?        00:00:00 watchdog/7
   42 ?        00:00:00 migration/7
   43 ?        00:00:00 ksoftirqd/7
   45 ?        00:00:00 kworker/7:0H
   46 ?        00:00:00 watchdog/8
   47 ?        00:00:00 migration/8
   48 ?        00:00:00 ksoftirqd/8
   50 ?        00:00:00 kworker/8:0H
   51 ?        00:00:00 watchdog/9
   52 ?        00:00:00 migration/9
   53 ?        00:00:00 ksoftirqd/9
   55 ?        00:00:00 kworker/9:0H
   56 ?        00:00:00 watchdog/10
   57 ?        00:00:00 migration/10
   58 ?        00:00:00 ksoftirqd/10
   60 ?        00:00:00 kworker/10:0H
   61 ?        00:00:00 watchdog/11
   62 ?        00:00:00 migration/11
   63 ?        00:00:00 ksoftirqd/11
   65 ?        00:00:00 kworker/11:0H
   66 ?        00:00:00 watchdog/12
   67 ?        00:00:00 migration/12
   68 ?        00:00:00 ksoftirqd/12
   69 ?        00:00:00 kworker/12:0
   70 ?        00:00:00 kworker/12:0H
   71 ?        00:00:00 watchdog/13
   72 ?        00:00:00 migration/13
   73 ?        00:00:00 ksoftirqd/13
   75 ?        00:00:00 kworker/13:0H
   76 ?        00:00:00 watchdog/14
   77 ?        00:00:00 migration/14
   78 ?        00:00:00 ksoftirqd/14
   80 ?        00:00:00 kworker/14:0H
   81 ?        00:00:00 watchdog/15
   82 ?        00:00:00 migration/15
   83 ?        00:00:00 ksoftirqd/15
   85 ?        00:00:00 kworker/15:0H
   86 ?        00:00:00 kdevtmpfs
   87 ?        00:00:00 netns
   88 ?        00:00:00 khungtaskd
   89 ?        00:00:00 writeback
   90 ?        00:00:00 ksmd
   91 ?        00:00:00 khugepaged
   92 ?        00:00:00 crypto
   93 ?        00:00:00 bioset
   94 ?        00:00:00 kblockd
   97 ?        00:00:00 kswapd0
   98 ?        00:00:00 vmstat
  107 ?        00:00:00 kthrotld
  108 ?        00:00:00 acpi_thermal_pm
  109 ?        00:00:00 ipv6_addrconf
  110 ?        00:00:00 deferwq
  151 ?        00:00:00 kworker/8:1
  155 ?        00:00:00 kworker/14:1
  156 ?        00:00:00 kworker/15:1
  239 ?        00:00:00 kauditd
  371 ?        00:00:00 edac-poller
  372 ?        00:00:00 cfg80211
  374 ?        00:00:00 kipmi0
  376 ?        00:00:00 ata_sff
  377 ?        00:00:00 nvme
  414 ?        00:00:00 scsi_eh_0
  415 ?        00:00:00 scsi_tmf_0
  416 ?        00:00:00 scsi_eh_1
  417 ?        00:00:00 scsi_tmf_1
  418 ?        00:00:00 scsi_eh_2
  419 ?        00:00:00 bioset
  420 ?        00:00:00 scsi_tmf_2
  421 ?        00:00:00 nvme
  422 ?        00:00:00 bioset
  423 ?        00:00:00 scsi_eh_3
  424 ?        00:00:00 scsi_tmf_3
  425 ?        00:00:00 scsi_eh_4
  426 ?        00:00:00 scsi_tmf_4
  427 ?        00:00:00 scsi_eh_5
  428 ?        00:00:00 scsi_tmf_5
  448 ?        00:00:00 kvm-irqfd-clean
  459 ?        00:00:00 bioset
  460 ?        00:00:00 scsi_eh_6
  461 ?        00:00:00 scsi_tmf_6
  462 ?        00:00:00 usb-storage
  463 ?        00:00:00 bioset
  464 ?        00:00:00 bioset
  469 ?        00:00:00 kworker/0:1H
  470 ?        00:00:00 scsi_eh_7
  471 ?        00:00:00 scsi_tmf_7
  472 ?        00:00:02 usb-storage
  473 ?        00:00:00 kworker/7:1H
  488 ?        00:00:00 bioset
  497 ?        00:00:00 bioset
  511 ?        00:00:00 kworker/1:1H
  512 ?        00:00:00 kworker/2:1H
  514 ?        00:00:00 kworker/3:1H
  515 ?        00:00:00 kworker/4:1H
  516 ?        00:00:00 kworker/8:1H
  531 ?        00:00:00 kworker/5:1H
  532 ?        00:00:00 kworker/12:1H
  553 ?        00:00:00 jbd2/sdd9-8
  554 ?        00:00:00 ext4-rsv-conver
  558 ?        00:00:00 ext4-rsv-conver
  637 ?        00:00:00 kworker/6:1H
  649 ?        00:00:00 kworker/9:1H
  653 ?        00:00:00 spl_kmem_cache
  654 ?        00:00:00 spl_kmem_cache
  655 ?        00:00:00 spl_kmem_cache
  656 ?        00:00:00 spl_kmem_cache
  657 ?        00:00:00 spl_system_task
  658 ?        00:00:00 spl_system_task
  659 ?        00:00:00 spl_system_task
  660 ?        00:00:00 spl_system_task
  661 ?        00:00:00 spl_system_task
  662 ?        00:00:00 spl_system_task
  664 ?        00:00:00 spl_system_task
  665 ?        00:00:00 spl_system_task
  666 ?        00:00:00 spl_system_task
  667 ?        00:00:00 spl_system_task
  668 ?        00:00:00 spl_system_task
  669 ?        00:00:00 spl_system_task
  670 ?        00:00:00 spl_system_task
  671 ?        00:00:00 spl_system_task
  672 ?        00:00:00 spl_system_task
  673 ?        00:00:00 spl_system_task
  674 ?        00:00:00 spl_system_task
  675 ?        00:00:00 spl_system_task
  676 ?        00:00:00 spl_system_task
  677 ?        00:00:00 spl_system_task
  678 ?        00:00:00 spl_system_task
  679 ?        00:00:00 spl_system_task
  680 ?        00:00:00 spl_system_task
  681 ?        00:00:00 spl_system_task
  682 ?        00:00:00 spl_system_task
  683 ?        00:00:00 spl_system_task
  684 ?        00:00:00 spl_system_task
  685 ?        00:00:00 spl_system_task
  686 ?        00:00:00 spl_system_task
  687 ?        00:00:00 spl_system_task
  688 ?        00:00:00 spl_system_task
  689 ?        00:00:00 spl_system_task
  690 ?        00:00:00 spl_system_task
  691 ?        00:00:00 spl_system_task
  692 ?        00:00:00 spl_system_task
  693 ?        00:00:00 spl_system_task
  694 ?        00:00:00 spl_system_task
  695 ?        00:00:00 spl_system_task
  696 ?        00:00:00 spl_system_task
  697 ?        00:00:00 spl_system_task
  698 ?        00:00:00 spl_system_task
  699 ?        00:00:00 spl_system_task
  700 ?        00:00:00 spl_system_task
  701 ?        00:00:00 spl_system_task
  702 ?        00:00:00 spl_system_task
  703 ?        00:00:00 spl_system_task
  704 ?        00:00:00 spl_system_task
  705 ?        00:00:00 spl_system_task
  706 ?        00:00:00 spl_system_task
  707 ?        00:00:00 spl_system_task
  708 ?        00:00:00 spl_system_task
  709 ?        00:00:00 spl_system_task
  710 ?        00:00:00 spl_system_task
  711 ?        00:00:00 spl_system_task
  712 ?        00:00:00 spl_system_task
  713 ?        00:00:00 spl_system_task
  714 ?        00:00:00 spl_system_task
  715 ?        00:00:00 spl_system_task
  716 ?        00:00:00 spl_system_task
  717 ?        00:00:00 spl_system_task
  718 ?        00:00:00 spl_system_task
  719 ?        00:00:00 spl_system_task
  720 ?        00:00:00 spl_system_task
  721 ?        00:00:00 spl_system_task
  722 ?        00:00:00 spl_dynamic_tas
  731 ?        00:00:02 dbu_evict
  732 ?        00:00:00 arc_prune
  733 ?        00:00:00 arc_prune
  734 ?        00:00:00 arc_prune
  735 ?        00:00:00 arc_prune
  736 ?        00:00:00 arc_prune
  737 ?        00:00:00 arc_prune
  738 ?        00:00:00 arc_prune
  739 ?        00:00:00 arc_prune
  740 ?        00:00:00 arc_prune
  741 ?        00:00:00 arc_prune
  742 ?        00:00:00 arc_prune
  743 ?        00:00:00 arc_prune
  744 ?        00:00:00 arc_prune
  745 ?        00:00:00 arc_prune
  746 ?        00:00:00 arc_prune
  747 ?        00:00:00 arc_prune
  748 ?        00:00:00 arc_reclaim
  749 ?        00:00:00 arc_user_evicts
  750 ?        00:00:02 l2arc_feed
  751 ?        00:00:00 z_unmount
  770 ?        00:00:00 kworker/11:1H
  771 ?        00:00:00 kworker/11:2
  833 ?        00:00:00 kworker/13:1H
  834 ?        00:00:00 kworker/10:1H
  861 ?        00:00:00 kworker/14:1H
  869 ?        00:00:00 jbd2/sdd6-8
  870 ?        00:00:00 ext4-rsv-conver
  876 ?        00:00:00 kworker/15:1H
 1623 ?        00:00:02 z_null_iss
 1624 ?        00:00:00 z_null_int
 1625 ?        00:00:00 z_rd_iss
 1626 ?        00:00:00 z_rd_iss
 1627 ?        00:00:00 z_rd_iss
 1628 ?        00:00:00 z_rd_iss
 1629 ?        00:00:00 z_rd_iss
 1630 ?        00:00:00 z_rd_iss
 1631 ?        00:00:00 z_rd_iss
 1632 ?        00:00:00 z_rd_iss
 1633 ?        00:00:00 z_rd_int_0
 1634 ?        00:00:00 z_rd_int_0
 1635 ?        00:00:00 z_rd_int_0
 1636 ?        00:00:00 z_rd_int_0
 1637 ?        00:00:00 z_rd_int_0
 1638 ?        00:00:00 z_rd_int_0
 1639 ?        00:00:00 z_rd_int_0
 1640 ?        00:00:00 z_rd_int_0
 1641 ?        00:00:00 z_rd_int_0
 1642 ?        00:00:00 z_rd_int_0
 1643 ?        00:00:00 z_rd_int_0
 1644 ?        00:00:00 z_rd_int_0
 1645 ?        00:00:00 z_rd_int_1
 1646 ?        00:00:00 z_rd_int_1
 1647 ?        00:00:00 z_rd_int_1
 1648 ?        00:00:00 z_rd_int_1
 1649 ?        00:00:00 z_rd_int_1
 1650 ?        00:00:00 z_rd_int_1
 1651 ?        00:00:00 z_rd_int_1
 1652 ?        00:00:00 z_rd_int_1
 1653 ?        00:00:00 z_rd_int_1
 1654 ?        00:00:00 z_rd_int_1
 1655 ?        00:00:00 z_rd_int_1
 1656 ?        00:00:00 z_rd_int_1
 1657 ?        00:00:00 z_rd_int_2
 1658 ?        00:00:00 z_rd_int_2
 1659 ?        00:00:00 z_rd_int_2
 1660 ?        00:00:00 z_rd_int_2
 1661 ?        00:00:00 z_rd_int_2
 1662 ?        00:00:00 z_rd_int_2
 1663 ?        00:00:00 z_rd_int_2
 1664 ?        00:00:00 z_rd_int_2
 1665 ?        00:00:00 z_rd_int_2
 1666 ?        00:00:00 z_rd_int_2
 1667 ?        00:00:00 z_rd_int_2
 1668 ?        00:00:00 z_rd_int_2
 1669 ?        00:00:00 z_rd_int_3
 1670 ?        00:00:00 z_rd_int_3
 1671 ?        00:00:00 z_rd_int_3
 1672 ?        00:00:00 z_rd_int_3
 1673 ?        00:00:00 z_rd_int_3
 1674 ?        00:00:00 z_rd_int_3
 1675 ?        00:00:00 z_rd_int_3
 1676 ?        00:00:00 z_rd_int_3
 1677 ?        00:00:00 z_rd_int_3
 1678 ?        00:00:00 z_rd_int_3
 1679 ?        00:00:00 z_rd_int_3
 1680 ?        00:00:00 z_rd_int_3
 1681 ?        00:00:00 z_rd_int_4
 1682 ?        00:00:00 z_rd_int_4
 1683 ?        00:00:00 z_rd_int_4
 1684 ?        00:00:00 z_rd_int_4
 1685 ?        00:00:00 z_rd_int_4
 1686 ?        00:00:00 z_rd_int_4
 1687 ?        00:00:00 z_rd_int_4
 1688 ?        00:00:00 z_rd_int_4
 1689 ?        00:00:00 z_rd_int_4
 1690 ?        00:00:00 z_rd_int_4
 1691 ?        00:00:00 z_rd_int_4
 1692 ?        00:00:00 z_rd_int_4
 1693 ?        00:00:00 z_rd_int_5
 1694 ?        00:00:00 z_rd_int_5
 1695 ?        00:00:00 z_rd_int_5
 1696 ?        00:00:00 z_rd_int_5
 1697 ?        00:00:00 z_rd_int_5
 1698 ?        00:00:00 z_rd_int_5
 1699 ?        00:00:00 z_rd_int_5
 1700 ?        00:00:00 z_rd_int_5
 1701 ?        00:00:00 z_rd_int_5
 1702 ?        00:00:00 z_rd_int_5
 1703 ?        00:00:00 z_rd_int_5
 1704 ?        00:00:00 z_rd_int_5
 1705 ?        00:00:00 z_rd_int_6
 1706 ?        00:00:00 z_rd_int_6
 1707 ?        00:00:00 z_rd_int_6
 1708 ?        00:00:00 z_rd_int_6
 1709 ?        00:00:00 z_rd_int_6
 1710 ?        00:00:00 z_rd_int_6
 1711 ?        00:00:00 z_rd_int_6
 1712 ?        00:00:00 z_rd_int_6
 1713 ?        00:00:00 z_rd_int_6
 1714 ?        00:00:00 z_rd_int_6
 1715 ?        00:00:00 z_rd_int_6
 1716 ?        00:00:00 z_rd_int_6
 1717 ?        00:00:00 z_rd_int_7
 1718 ?        00:00:00 z_rd_int_7
 1719 ?        00:00:00 z_rd_int_7
 1720 ?        00:00:00 z_rd_int_7
 1721 ?        00:00:00 z_rd_int_7
 1722 ?        00:00:00 z_rd_int_7
 1723 ?        00:00:00 z_rd_int_7
 1724 ?        00:00:00 z_rd_int_7
 1725 ?        00:00:00 z_rd_int_7
 1726 ?        00:00:00 z_rd_int_7
 1727 ?        00:00:00 z_rd_int_7
 1728 ?        00:00:00 z_rd_int_7
 1729 ?        00:00:04 z_wr_iss
 1730 ?        00:00:04 z_wr_iss
 1731 ?        00:00:04 z_wr_iss
 1732 ?        00:00:04 z_wr_iss
 1733 ?        00:00:04 z_wr_iss
 1734 ?        00:00:04 z_wr_iss
 1735 ?        00:00:04 z_wr_iss
 1736 ?        00:00:04 z_wr_iss
 1737 ?        00:00:04 z_wr_iss
 1738 ?        00:00:04 z_wr_iss
 1739 ?        00:00:04 z_wr_iss
 1740 ?        00:00:04 z_wr_iss
 1741 ?        00:00:00 z_wr_iss_h
 1742 ?        00:00:00 z_wr_iss_h
 1743 ?        00:00:00 z_wr_iss_h
 1744 ?        00:00:00 z_wr_iss_h
 1745 ?        00:00:00 z_wr_iss_h
 1746 ?        00:00:00 z_wr_int_0
 1747 ?        00:00:00 z_wr_int_0
 1748 ?        00:00:00 z_wr_int_0
 1749 ?        00:00:00 z_wr_int_0
 1750 ?        00:00:00 z_wr_int_0
 1751 ?        00:00:00 z_wr_int_0
 1752 ?        00:00:00 z_wr_int_0
 1753 ?        00:00:00 z_wr_int_0
 1754 ?        00:00:00 z_wr_int_0
 1755 ?        00:00:00 z_wr_int_0
 1756 ?        00:00:00 z_wr_int_0
 1757 ?        00:00:00 z_wr_int_0
 1758 ?        00:00:00 z_wr_int_1
 1759 ?        00:00:00 z_wr_int_1
 1760 ?        00:00:00 z_wr_int_1
 1761 ?        00:00:00 z_wr_int_1
 1762 ?        00:00:00 z_wr_int_1
 1763 ?        00:00:00 z_wr_int_1
 1764 ?        00:00:00 z_wr_int_1
 1765 ?        00:00:00 z_wr_int_1
 1766 ?        00:00:00 z_wr_int_1
 1767 ?        00:00:00 z_wr_int_1
 1768 ?        00:00:00 z_wr_int_1
 1769 ?        00:00:00 z_wr_int_1
 1770 ?        00:00:00 z_wr_int_2
 1771 ?        00:00:00 z_wr_int_2
 1772 ?        00:00:00 z_wr_int_2
 1773 ?        00:00:00 z_wr_int_2
 1774 ?        00:00:00 z_wr_int_2
 1775 ?        00:00:00 z_wr_int_2
 1776 ?        00:00:00 z_wr_int_2
 1777 ?        00:00:00 z_wr_int_2
 1778 ?        00:00:00 z_wr_int_2
 1779 ?        00:00:00 z_wr_int_2
 1780 ?        00:00:00 z_wr_int_2
 1781 ?        00:00:00 z_wr_int_2
 1782 ?        00:00:00 z_wr_int_3
 1783 ?        00:00:00 z_wr_int_3
 1784 ?        00:00:00 z_wr_int_3
 1785 ?        00:00:00 z_wr_int_3
 1786 ?        00:00:00 z_wr_int_3
 1787 ?        00:00:00 z_wr_int_3
 1788 ?        00:00:00 z_wr_int_3
 1789 ?        00:00:00 z_wr_int_3
 1790 ?        00:00:00 z_wr_int_3
 1791 ?        00:00:00 z_wr_int_3
 1792 ?        00:00:00 z_wr_int_3
 1793 ?        00:00:00 z_wr_int_3
 1794 ?        00:00:00 z_wr_int_4
 1795 ?        00:00:00 z_wr_int_4
 1796 ?        00:00:00 z_wr_int_4
 1797 ?        00:00:00 z_wr_int_4
 1798 ?        00:00:00 z_wr_int_4
 1799 ?        00:00:00 z_wr_int_4
 1800 ?        00:00:00 z_wr_int_4
 1801 ?        00:00:00 z_wr_int_4
 1802 ?        00:00:00 z_wr_int_4
 1803 ?        00:00:00 z_wr_int_4
 1804 ?        00:00:00 z_wr_int_4
 1805 ?        00:00:00 z_wr_int_4
 1806 ?        00:00:00 z_wr_int_5
 1807 ?        00:00:00 z_wr_int_5
 1808 ?        00:00:00 z_wr_int_5
 1809 ?        00:00:00 z_wr_int_5
 1810 ?        00:00:00 z_wr_int_5
 1811 ?        00:00:00 z_wr_int_5
 1812 ?        00:00:00 z_wr_int_5
 1813 ?        00:00:00 z_wr_int_5
 1814 ?        00:00:00 z_wr_int_5
 1815 ?        00:00:00 z_wr_int_5
 1816 ?        00:00:00 z_wr_int_5
 1817 ?        00:00:00 z_wr_int_5
 1818 ?        00:00:00 z_wr_int_6
 1819 ?        00:00:00 z_wr_int_6
 1820 ?        00:00:00 z_wr_int_6
 1821 ?        00:00:00 z_wr_int_6
 1822 ?        00:00:00 z_wr_int_6
 1823 ?        00:00:00 z_wr_int_6
 1824 ?        00:00:00 z_wr_int_6
 1825 ?        00:00:00 z_wr_int_6
 1826 ?        00:00:00 z_wr_int_6
 1827 ?        00:00:00 z_wr_int_6
 1828 ?        00:00:00 z_wr_int_6
 1829 ?        00:00:00 z_wr_int_6
 1830 ?        00:00:00 z_wr_int_7
 1831 ?        00:00:00 z_wr_int_7
 1832 ?        00:00:00 z_wr_int_7
 1833 ?        00:00:00 z_wr_int_7
 1834 ?        00:00:00 z_wr_int_7
 1835 ?        00:00:00 z_wr_int_7
 1836 ?        00:00:00 z_wr_int_7
 1837 ?        00:00:00 z_wr_int_7
 1838 ?        00:00:00 z_wr_int_7
 1839 ?        00:00:00 z_wr_int_7
 1840 ?        00:00:00 z_wr_int_7
 1841 ?        00:00:00 z_wr_int_7
 1842 ?        00:00:00 z_wr_int_h
 1843 ?        00:00:00 z_wr_int_h
 1844 ?        00:00:00 z_wr_int_h
 1845 ?        00:00:00 z_wr_int_h
 1846 ?        00:00:00 z_wr_int_h
 1847 ?        00:00:00 z_fr_iss_0
 1848 ?        00:00:00 z_fr_iss_0
 1849 ?        00:00:00 z_fr_iss_0
 1850 ?        00:00:00 z_fr_iss_0
 1851 ?        00:00:00 z_fr_iss_0
 1852 ?        00:00:00 z_fr_iss_0
 1853 ?        00:00:00 z_fr_iss_0
 1854 ?        00:00:00 z_fr_iss_0
 1855 ?        00:00:00 z_fr_iss_0
 1856 ?        00:00:00 z_fr_iss_0
 1857 ?        00:00:00 z_fr_iss_0
 1858 ?        00:00:00 z_fr_iss_0
 1859 ?        00:00:00 z_fr_iss_1
 1860 ?        00:00:00 z_fr_iss_1
 1861 ?        00:00:00 z_fr_iss_1
 1862 ?        00:00:00 z_fr_iss_1
 1863 ?        00:00:00 z_fr_iss_1
 1864 ?        00:00:00 z_fr_iss_1
 1865 ?        00:00:00 z_fr_iss_1
 1866 ?        00:00:00 z_fr_iss_1
 1867 ?        00:00:00 z_fr_iss_1
 1868 ?        00:00:00 z_fr_iss_1
 1869 ?        00:00:00 z_fr_iss_1
 1870 ?        00:00:00 z_fr_iss_1
 1871 ?        00:00:00 z_fr_iss_2
 1872 ?        00:00:00 z_fr_iss_2
 1873 ?        00:00:00 z_fr_iss_2
 1874 ?        00:00:00 z_fr_iss_2
 1875 ?        00:00:00 z_fr_iss_2
 1876 ?        00:00:00 z_fr_iss_2
 1877 ?        00:00:00 z_fr_iss_2
 1878 ?        00:00:00 z_fr_iss_2
 1879 ?        00:00:00 z_fr_iss_2
 1880 ?        00:00:00 z_fr_iss_2
 1881 ?        00:00:00 z_fr_iss_2
 1882 ?        00:00:00 z_fr_iss_2
 1883 ?        00:00:00 z_fr_iss_3
 1884 ?        00:00:00 z_fr_iss_3
 1885 ?        00:00:00 z_fr_iss_3
 1886 ?        00:00:00 z_fr_iss_3
 1887 ?        00:00:00 z_fr_iss_3
 1888 ?        00:00:00 z_fr_iss_3
 1889 ?        00:00:00 z_fr_iss_3
 1890 ?        00:00:00 z_fr_iss_3
 1891 ?        00:00:00 z_fr_iss_3
 1892 ?        00:00:00 z_fr_iss_3
 1893 ?        00:00:00 z_fr_iss_3
 1894 ?        00:00:00 z_fr_iss_3
 1895 ?        00:00:00 z_fr_iss_4
 1896 ?        00:00:00 z_fr_iss_4
 1897 ?        00:00:00 z_fr_iss_4
 1898 ?        00:00:00 z_fr_iss_4
 1899 ?        00:00:00 z_fr_iss_4
 1900 ?        00:00:00 z_fr_iss_4
 1901 ?        00:00:00 z_fr_iss_4
 1902 ?        00:00:00 z_fr_iss_4
 1903 ?        00:00:00 z_fr_iss_4
 1904 ?        00:00:00 z_fr_iss_4
 1905 ?        00:00:00 z_fr_iss_4
 1906 ?        00:00:00 z_fr_iss_4
 1907 ?        00:00:00 z_fr_iss_5
 1908 ?        00:00:00 z_fr_iss_5
 1909 ?        00:00:00 z_fr_iss_5
 1910 ?        00:00:00 z_fr_iss_5
 1911 ?        00:00:00 z_fr_iss_5
 1912 ?        00:00:00 z_fr_iss_5
 1913 ?        00:00:00 z_fr_iss_5
 1914 ?        00:00:00 z_fr_iss_5
 1915 ?        00:00:00 z_fr_iss_5
 1916 ?        00:00:00 z_fr_iss_5
 1917 ?        00:00:00 z_fr_iss_5
 1918 ?        00:00:00 z_fr_iss_5
 1919 ?        00:00:00 z_fr_iss_6
 1920 ?        00:00:00 z_fr_iss_6
 1921 ?        00:00:00 z_fr_iss_6
 1922 ?        00:00:00 z_fr_iss_6
 1923 ?        00:00:00 z_fr_iss_6
 1924 ?        00:00:00 z_fr_iss_6
 1925 ?        00:00:00 z_fr_iss_6
 1926 ?        00:00:00 z_fr_iss_6
 1927 ?        00:00:00 z_fr_iss_6
 1928 ?        00:00:00 z_fr_iss_6
 1929 ?        00:00:00 z_fr_iss_6
 1930 ?        00:00:00 z_fr_iss_6
 1931 ?        00:00:00 z_fr_iss_7
 1932 ?        00:00:00 z_fr_iss_7
 1933 ?        00:00:00 z_fr_iss_7
 1934 ?        00:00:00 z_fr_iss_7
 1935 ?        00:00:00 z_fr_iss_7
 1936 ?        00:00:00 z_fr_iss_7
 1937 ?        00:00:00 z_fr_iss_7
 1938 ?        00:00:00 z_fr_iss_7
 1939 ?        00:00:00 z_fr_iss_7
 1940 ?        00:00:00 z_fr_iss_7
 1941 ?        00:00:00 z_fr_iss_7
 1942 ?        00:00:00 z_fr_iss_7
 1943 ?        00:00:00 z_fr_int
 1944 ?        00:00:00 z_cl_iss
 1945 ?        00:00:00 z_cl_int
 1946 ?        00:00:00 z_ioctl_iss
 1947 ?        00:00:00 z_ioctl_int
 1948 ?        00:00:00 z_zvol
 1949 ?        00:00:00 metaslab_group_
 1950 ?        00:00:00 metaslab_group_
 1951 ?        00:00:00 metaslab_group_
 1952 ?        00:00:00 metaslab_group_
 1953 ?        00:00:00 metaslab_group_
 1954 ?        00:00:00 metaslab_group_
 1955 ?        00:00:00 metaslab_group_
 1956 ?        00:00:00 metaslab_group_
 1961 ?        00:00:00 z_iput
 1962 ?        00:00:00 z_iput
 1963 ?        00:00:00 z_iput
 1964 ?        00:00:00 z_iput
 1965 ?        00:00:00 z_iput
 1966 ?        00:00:00 z_iput
 1967 ?        00:00:00 z_iput
 1968 ?        00:00:00 z_iput
 1969 ?        00:00:00 z_iput
 1970 ?        00:00:00 z_iput
 1971 ?        00:00:00 z_iput
 1972 ?        00:00:00 z_iput
 1973 ?        00:00:00 z_iput
 1974 ?        00:00:00 z_iput
 1975 ?        00:00:00 z_iput
 1976 ?        00:00:00 z_iput
 2012 ?        00:00:00 txg_quiesce
 2013 ?        00:00:06 txg_sync
 2021 ?        00:00:00 zil_clean
 2025 ?        00:00:00 zil_clean
 2029 ?        00:00:00 zil_clean
 2033 ?        00:00:00 zil_clean
 2308 ?        00:00:00 kworker/13:2
 2910 ?        00:00:00 zil_clean
 3304 ?        00:00:00 zil_clean
 3440 ?        00:00:00 zil_clean
 3529 ?        00:00:00 zil_clean
 3602 ?        00:00:00 zil_clean
 3699 ?        00:00:00 zil_clean
 3774 ?        00:00:00 zil_clean
 3947 ?        00:00:00 zil_clean
 4313 ?        00:00:00 zil_clean
 4465 ?        00:00:00 zil_clean
 4574 ?        00:00:00 zil_clean
 4945 ?        00:00:00 zil_clean
 5457 ?        00:00:00 zil_clean
 5648 ?        00:00:00 zil_clean
 5731 ?        00:00:00 zil_clean
 6159 ?        00:00:00 zil_clean
 7474 ?        00:00:00 zil_clean
 8440 ?        00:00:00 kworker/7:0
 8467 ?        00:00:00 kworker/u32:1
 8544 ?        00:00:00 kworker/9:0
 9073 ?        00:00:00 kworker/2:0
10358 ?        00:00:00 kworker/3:0
10428 ?        00:00:00 kworker/10:0
11252 ?        00:00:00 kworker/1:2
11289 ?        00:00:00 kworker/6:0
11626 ?        00:00:00 kworker/0:1
11657 ?        00:00:00 kworker/5:1
12222 ?        00:00:00 kworker/7:1
12273 ?        00:00:00 kworker/11:0
12455 ?        00:00:00 kworker/4:1
12528 ?        00:00:00 kworker/9:1
12690 ?        00:00:00 kworker/14:0
12793 ?        00:00:00 kworker/13:0
13352 ?        00:00:00 kworker/8:2
13643 ?        00:00:00 kworker/2:1
13996 ?        00:00:00 kworker/3:2
14300 ?        00:00:00 kworker/15:2
14469 ?        00:00:00 kworker/12:1
14718 ?        00:00:00 kworker/6:1
14822 ?        00:00:00 kworker/5:2
14847 ?        00:00:00 kworker/0:2
14867 ?        00:00:00 kworker/1:0
15446 ?        00:00:00 kworker/7:2
16009 ?        00:00:00 kworker/4:2
16316 ?        00:00:00 kworker/10:1
16579 ?        00:00:00 kworker/13:1
17066 ?        00:00:00 kworker/2:2
17172 ?        00:00:00 kworker/11:1
17463 ?        00:00:00 kworker/3:1
17475 ?        00:00:00 kworker/u32:2
17921 ?        00:00:00 kworker/6:2
17933 ?        00:00:00 kworker/u32:3
17934 ?        00:00:00 kworker/u32:4
17935 ?        00:00:00 kworker/u32:5
17936 ?        00:00:00 kworker/u32:6
18184 ?        00:00:00 kworker/u32:7
18237 ?        00:00:00 kworker/14:2
18335 ?        00:00:00 kworker/5:0
18387 ?        00:00:00 kworker/0:0
18400 ?        00:00:00 kworker/5:3
18402 ?        00:00:00 kworker/5:4
18438 ?        00:00:00 kworker/6:3
18494 ?        00:00:00 kworker/3:3
18531 ?        00:00:00 kworker/8:0
18679 ?        00:00:00 kworker/1:1
18735 ?        00:00:00 kworker/9:2
19310 ?        00:00:00 kworker/12:2
19664 ?        00:00:00 kworker/4:0
20137 ?        00:00:00 zil_clean
20271 ?        00:00:00 ath9k-hwrng
```

**Comment by behlendorf** [comment]
@kdomanski yes, that's normal.  For the 0.6.5.x releases those threads will be created when the pool is imported, for changes on the master branch those threads will be created and destroyed dynamically as needed.

**Comment by kdomanski** [comment]
Here's a task dump from a machine in a faulty state:
https://gist.githubusercontent.com/kdomanski/018813a8485ca86c7139411266d5b0da/raw/b9a5a55dcf0880ce488cd3786967ff0e88aac7a1/gistfile1.txt

Trying to reboot right after this dump caused the machine to block on the systemd task.
Console shows `Reached target shutdown` - normally right after this I should get a split-second message about the drives being deactivated/spun down and then the machine should actually reboot.

Instead the "task blocked" message keeps appearing every 2 minutes and I can head the drives working. This state continues indefinitely.

**Comment by HankB** [comment]
I just hit what appears to be this issue on Ubuntu 16.04 running 0.6.5.6. It occurred while using rsync to copy a file system from another machine. I had previously done this and found that the result seemed to use a lot more disk space (--ashift=12, raidz2) so I destroyed the RAID and recreated with --ashift=9. Drives are modern 4K format. I saved a copy of 'dmesg' before rebooting. I also found that the network connection to the machine was down.

I don't know if this is unexpected or some kind of fluke. If there is any information I could provide should this happen again please let me know. It has only happened once.
First of three occurrences looks like
```
[43498.263473] INFO: task txg_sync:1911 blocked for more than 120 seconds.
[43498.263484]       Tainted: P           O    4.10.0-40-generic #44~16.04.1-Ubuntu
[43498.263489] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[43498.263494] txg_sync        D    0  1911      2 0x00000000
[43498.263501] Call Trace:
[43498.263514]  __schedule+0x232/0x700
[43498.263648]  ? zio_taskq_member.isra.8.constprop.16+0x70/0x70 [zfs]
[43498.263653]  schedule+0x36/0x80
[43498.263658]  schedule_timeout+0x235/0x3f0
[43498.263752]  ? zio_taskq_dispatch+0x94/0xa0 [zfs]
[43498.263845]  ? zio_issue_async+0x12/0x20 [zfs]
[43498.263937]  ? zio_nowait+0x79/0xf0 [zfs]
[43498.263943]  io_schedule_timeout+0xa4/0x110
[43498.263966]  cv_wait_common+0xbc/0x140 [spl]
[43498.263975]  ? wake_atomic_t_function+0x60/0x60
[43498.263988]  __cv_wait_io+0x18/0x20 [spl]
[43498.264080]  zio_wait+0xc4/0x160 [zfs]
[43498.264165]  dsl_pool_sync+0xb8/0x460 [zfs]
[43498.264257]  spa_sync+0x374/0xb40 [zfs]
[43498.264262]  ? default_wake_function+0x12/0x20
[43498.264356]  txg_sync_thread+0x3bf/0x630 [zfs]
[43498.264451]  ? txg_quiesce_thread+0x3f0/0x3f0 [zfs]
[43498.264463]  thread_generic_wrapper+0x72/0x80 [spl]
[43498.264469]  kthread+0x109/0x140
[43498.264481]  ? __thread_exit+0x20/0x20 [spl]
[43498.264486]  ? kthread_create_on_node+0x60/0x60
[43498.264492]  ret_from_fork+0x2c/0x40
```

**Comment by error10** [comment]
Looks like a duplicate of #2934.

**Comment by Kelvin3388** [comment]
Experiencing similar issue.

Linux version 3.10.0-693.5.2.el7_lustre.x86_64
SPL: Loaded module v0.7.3-1
ZFS: Loaded module v0.7.3-1


Feb 13 23:47:24 zfswh2-oss-17-2 kernel: INFO: task txg_sync:32290 blocked for more than 120 seconds.
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: txg_sync        D 0000000000000001     0 32290      2 0x00000080
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: ffff880fabf6bac0 0000000000000046 ffff880fb0723f40 ffff880fabf6bfd8
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: ffff880fabf6bfd8 ffff880fabf6bfd8 ffff880fb0723f40 ffff88207ec96cc0
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: 0000000000000000 7fffffffffffffff ffff881e9f808880 0000000000000001
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: Call Trace:
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff816a9569>] schedule+0x29/0x70
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff816a7079>] schedule_timeout+0x239/0x2c0
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0b9611f>] ? zio_taskq_dispatch+0x8f/0xa0 [zfs]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0b96162>] ? zio_issue_async+0x12/0x20 [zfs]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0b9a9dc>] ? zio_nowait+0xbc/0x150 [zfs]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff816a8bed>] io_schedule_timeout+0xad/0x130
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff810b1616>] ? prepare_to_wait_exclusive+0x56/0x90
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff816a8c88>] io_schedule+0x18/0x20
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0756462>] cv_wait_common+0xb2/0x150 [spl]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff810b1920>] ? wake_up_atomic_t+0x30/0x30
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0756558>] __cv_wait_io+0x18/0x20 [spl]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0b9a2ab>] zio_wait+0x10b/0x1b0 [zfs]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0b1053f>] dsl_pool_sync+0xbf/0x440 [zfs]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0b2e647>] spa_sync+0x437/0xdf0 [zfs]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff810c4832>] ? default_wake_function+0x12/0x20
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff810bd514>] ? __wake_up+0x44/0x50
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0b42881>] txg_sync_thread+0x301/0x510 [zfs]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0b42580>] ? txg_fini+0x2a0/0x2a0 [zfs]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0750fa1>] thread_generic_wrapper+0x71/0x80 [spl]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffffc0750f30>] ? __thread_exit+0x20/0x20 [spl]
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff810b099f>] kthread+0xcf/0xe0
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff8108ddfb>] ? do_exit+0x6bb/0xa40
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff810b08d0>] ? insert_kthread_work+0x40/0x40
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff816b4fd8>] ret_from_fork+0x58/0x90
Feb 13 23:47:24 zfswh2-oss-17-2 kernel: [<ffffffff810b08d0>] ? insert_kthread_work+0x40/0x40

**Comment by vukomir** [comment]
Experiencing similar issue
Linux Version 4.4.0-127-generic #153-Ubuntu x86_64
ZFS: 0.6.5.6-0ubuntu21

[704160.346088] INFO: task txg_sync:2271 blocked for more than 120 seconds.
[704160.346157]       Tainted: P           O    4.4.0-127-generic #153-Ubuntu
[704160.346206] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[704160.346257] txg_sync        D ffff881f965e3aa8     0  2271      2 0x00000000
[704160.346261]  ffff881f965e3aa8 ffff881f965e3a88 ffffffff81e13500 ffff881f972e8e00
[704160.346264]  ffff881f965e4000 ffff881fbd6172c0 7fffffffffffffff ffff8818836c2668
[704160.346266]  0000000000000001 ffff881f965e3ac0 ffffffff8184cae5 0000000000000000
[704160.346268] Call Trace:
[704160.346278]  [<ffffffff8184cae5>] schedule+0x35/0x80
[704160.346280]  [<ffffffff8184fc76>] schedule_timeout+0x1b6/0x270
[704160.346284]  [<ffffffff810bd5dc>] ? set_next_entity+0x9c/0xb0
[704160.346288]  [<ffffffff8102d650>] ? __switch_to+0x1c0/0x5d0
[704160.346292]  [<ffffffff810f92ae>] ? ktime_get+0x3e/0xb0
[704160.346295]  [<ffffffff8184c254>] io_schedule_timeout+0xa4/0x110
[704160.346304]  [<ffffffffc0205c4c>] cv_wait_common+0xbc/0x140 [spl]
[704160.346307]  [<ffffffff810c6f70>] ? wake_atomic_t_function+0x60/0x60
[704160.346311]  [<ffffffffc0205d28>] __cv_wait_io+0x18/0x20 [spl]
[704160.346352]  [<ffffffffc0365634>] zio_wait+0x114/0x200 [zfs]
[704160.346374]  [<ffffffffc02eed88>] dsl_pool_sync+0xb8/0x430 [zfs]
[704160.346400]  [<ffffffffc030a7b6>] spa_sync+0x366/0xb30 [zfs]
[704160.346403]  [<ffffffff810af022>] ? default_wake_function+0x12/0x20
[704160.346430]  [<ffffffffc031bc4a>] txg_sync_thread+0x3ba/0x630 [zfs]
[704160.346455]  [<ffffffffc031b890>] ? txg_delay+0x180/0x180 [zfs]
[704160.346459]  [<ffffffffc0200de0>] ? __thread_exit+0x20/0x20 [spl]
[704160.346463]  [<ffffffffc0200e53>] thread_generic_wrapper+0x73/0x80 [spl]
[704160.346466]  [<ffffffff810a3487>] kthread+0xe7/0x100
[704160.346468]  [<ffffffff810a33a0>] ? kthread_create_on_node+0x1e0/0x1e0
[704160.346471]  [<ffffffff818510f5>] ret_from_fork+0x55/0x80
[704160.346473]  [<ffffffff810a33a0>] ? kthread_create_on_node+0x1e0/0x1e0

**Comment by kpande** [comment]
closing due to age, open a new issue with relevant information if you have continuing issues.

**Comment by fbacchella** [comment]
There is some activity from ZFS users complaining about that. 
The lack of activity is coming from ZFS developers who don't care about it.

**Comment by kpande** [comment]
hi, your issue is *incredibly* unlikely to be the same. there is more than 2 years in between the issues, there have been a lot of "issue creep" on this ticket, so the best course of action is to close *this* one and you can open a *new* issue. thanks, @fbacchella.

**Comment by mvalites** [comment]
+1
System versions:
```
mval@bucket:~$ uname -a
uname: Linux bucket 5.4.0-1023-raspi #26-Ubuntu SMP PREEMPT Thu Nov 12 14:58:33 UTC 2020 aarch64 aarch64 aarch64 GNU/Linux

mval@bucket:~$ zfs --version
zfs-0.8.3-1ubuntu12.5
zfs-kmod-0.8.3-1ubuntu12.4

mval@bucket:~$ lsb_release  -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.1 LTS
Release:	20.04
Codename:	focal
mval@bucket:~$ 
```
dmsg:
```
[254235.428842] INFO: task txg_sync:3202 blocked for more than 120 seconds.
[254235.435682]       Tainted: P         C OE     5.4.0-1023-raspi #26-Ubuntu
[254235.442717] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[254235.450788] txg_sync        D    0  3202      2 0x00000028
[254235.450798] Call trace:
[254235.450812]  __switch_to+0x108/0x1c0
[254235.450820]  __schedule+0x328/0x828
[254235.450824]  schedule+0x4c/0xe8
[254235.450857]  cv_wait_common+0x134/0x178 [spl]
[254235.450875]  __cv_wait+0x34/0x40 [spl]
[254235.451064]  arc_read+0x160/0xdd0 [zfs]
[254235.451181]  dsl_scan_visitbp+0x1c4/0x908 [zfs]
[254235.451307]  dsl_scan_visitbp+0x220/0x908 [zfs]
[254235.451420]  dsl_scan_visitbp+0x41c/0x908 [zfs]
[254235.451530]  dsl_scan_visitbp+0x220/0x908 [zfs]
[254235.451638]  dsl_scan_visitbp+0x220/0x908 [zfs]
[254235.451747]  dsl_scan_visitbp+0x220/0x908 [zfs]
[254235.451857]  dsl_scan_visitbp+0x220/0x908 [zfs]
[254235.451968]  dsl_scan_visitbp+0x220/0x908 [zfs]
[254235.452090]  dsl_scan_visitbp+0x5f4/0x908 [zfs]
[254235.452209]  dsl_scan_visit_rootbp+0xd0/0x118 [zfs]
[254235.452329]  dsl_scan_visitds+0x148/0x468 [zfs]
[254235.452438]  dsl_scan_visit+0x12c/0x2a0 [zfs]
[254235.452547]  dsl_scan_sync+0x408/0x7e8 [zfs]
[254235.452666]  spa_sync_iterate_to_convergence+0x12c/0x1f0 [zfs]
[254235.452837]  spa_sync+0x2ec/0x520 [zfs]
[254235.452970]  txg_sync_thread+0x244/0x2a0 [zfs]
[254235.452996]  thread_generic_wrapper+0x78/0xa8 [spl]
[254235.453007]  kthread+0x150/0x170
[254235.453013]  ret_from_fork+0x10/0x18
```
